gcloud/_helpers.py

```diff
@@ -302,14 +303,30 @@ def _total_seconds(offset):
 def _rfc3339_to_datetime(dt_str):
     """Convert a string to a native timestamp.
 
+    .. note::
+
+       Python datetimes do not support nanosecond precision;  this function
+       therefore truncates such values to microseconds.
+
     :type dt_str: str
     :param dt_str: The string to convert.
 
     :rtype: :class:`datetime.datetime`
     :returns: The datetime object created from the string.
     """
-    return datetime.datetime.strptime(
-        dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    try:
+        return datetime.datetime.strptime(
+            dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    except ValueError:  # maybe nanosecond resoultion
+        prefix, suffix = dt_str.split('.')
+        fraction, zulu = suffix[:-1], suffix[-1]
```

Comment:
What should we be doing with the contra-spec case?  Do you mean:

```python
    try:
        return datetime.datetime.strptime(
            dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
    except ValueError:  # maybe nanosecond resoultion
        if not dt_str.endswith('Z'):
            raise
```

The lines after that are just trying to get the nanoseconds part separated from the `YYYY-dd-mmTHH:MM:SS` part:  they discard the `Z` as extraneous.